anchore_engine/services/policy_engine/engine/policy/gates/malware.py

```diff
@@ -80,8 +86,8 @@ def prepare_context(self, image_obj, context):
             AnalysisArtifact.analyzer_type == "base",
         ).all()
 
-        # Don't use a dict comprehension since context.data may have other content (not usually, but possible)
-        for scanner in scans:
-            context.data[scanner.artifact_key] = scanner.json_value
+        context.data[MALWARE_CONTEXT_KEY] = {
+            scanner.artifact_key: scanner.json_value for scanner in scans
+        }
```

Comment:
is `.data` going to always be available at this point? I thought there was a chance that this could be missing (hence the `AttributeError` check?

Reply:
This is a good point, the old code was a bit redundant. The data property is part of the context object and should always be present. There is a bigger review/update to the gates/context code needed here to address it more systematically, but I can fix it in this one.